composer/models/huggingface.py

```diff
@@ -88,23 +87,33 @@ def forward(self, batch):
     def loss(self, outputs, batch):
         return outputs['loss']
 
-    def validate(self, batch):
+    def eval_forward(self, batch, outputs: Optional[Any] = None):
         if self.use_logits:
-            labels = batch.pop('labels')
-            output = self.forward(batch)
+            if outputs:
+                output = outputs
+            else:
+                output = self.forward(batch)
             output = output['logits']
 
             # if we are in the single class case, then remove the classes dimension
             if output.shape[1] == 1:
                 output = output.squeeze(dim=1)
 
-            return output, labels
+            return output
         else:
-            output = self.forward(batch)
-            return output, None
+            return outputs if outputs else self.forward(batch)
 
-    def metrics(self, train: bool = False):
-        return self.train_metrics if train else self.valid_metrics
+    def get_metrics(self, is_train: bool = False) -> Dict[str, Metric]:
+        if is_train:
+            metrics = deepcopy(self.train_metrics)
```

Comment:
why do we need to deepcopy here?

Reply:
if there are multiple evaluators, we want to make sure that each evaluator only updates its own set of metrics which are set using this method.

Reply:
ah, is this something we can handle internally? Users implementing their models may not realize they need to return a deepcopy.

Reply:
yep, good point -- will move to inside the trainer